Python sdk

LucasRoesler opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Create a python function SDK that contains standard utilities for accessing secrets and environment variables. 

TravisCI is also configured to test the python SDK on every PR and to automatically deploy on tags
![image](https://user-images.githubusercontent.com/891889/96372793-ee775100-1168-11eb-8a3e-06972343dbfb.png)

Note that _before_ tagging, someone must 

```sh
poetry version patch|minor|major|<value>
```
which is documented in the readme of function-sdk.

If this is not run and committed _before_ the tag, then the deployment will fail because PyPI is immutable, you can not push a release to an existing version.

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

#5 

## Which issue(s) this PR fixes 
<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged): -->
Fixes #5

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Includes unit tests

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Version change (see: Impact to existing users)

## Impact to existing users

<!-- If none, state why and how you can be sure of that. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

LucasRoesler committed:
Initial python sdk project

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

LucasRoesler committed:
Expand project setup and instructions

**What**
- Add more details to the project toml
- Add a very simple makefile
- Expand the instructions in the README to document how the project is
  setup, how to integrate with VSCode, and how to publish the package

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

LucasRoesler committed:
Add travis configuration

**What**
- Add .travis.yaml to enable CI/CD for the python sdk
- Add example usage to the python function-sdk README

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

alexellis commented on python/function-sdk/README.md:
````diff
@@ -0,0 +1,86 @@
+
+```py
+# handler.py
+from function_sdk.secrets import get_secret
````
From the call today - what about `openfaas-sdk` from @LucasRoesler ?

LucasRoesler committed:
Rename and Restructure python project

**What**
- Rename to the openfaas-sdk and then restructure so that imports now
  look like `from openfaas_sdk.function.secrets import get_secret`

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

LucasRoesler committed:
Improve usage examples

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

LucasRoesler commented:
@alexellis i have renamed everything as needed and updated the docs, take a look at the README for an example usage

you can also test this locally by installing it locally and then opening a python shell

```sh
$ cd python/openfaas-sdk/
$ poetry shell
$ poetry install 
$ pip install .
$ python
```

```py
from openfaas_sdk import function

help(function)
help(function.secrets)
help(function.env)
```

You can also go here https://test.pypi.org/project/openfaas-sdk/ to see what it will look like after it is published

LucasRoesler committed:
Fix lint stage definition in travisci

Signed-off-by: Lucas Roesler <roesler.lucas@gmail.com>

alexellis commented:
Discussed on the call today. The same for Node.js came up, which we may look at next.
